http-cors/src/main/scala/org/apache/pekko/http/cors/javadsl/settings/CorsSettings.scala

```diff
@@ -0,0 +1,59 @@
+package org.apache.pekko.http.cors.javadsl.settings
+
+import java.util.OptionalLong
+import org.apache.pekko
+import pekko.actor.ActorSystem
+import pekko.annotation.DoNotInherit
+import pekko.http.javadsl.model.HttpMethod
+import pekko.http.cors.javadsl.model.{ HttpHeaderRange, HttpOriginMatcher }
+import pekko.http.cors.scaladsl
+import pekko.http.cors.scaladsl.settings.CorsSettingsImpl
+import com.typesafe.config.Config
+
+/**
+ * Public API but not intended for subclassing
+ */
+@DoNotInherit
+abstract class CorsSettings { self: CorsSettingsImpl =>
+
+  def getAllowGenericHttpRequests: Boolean
+  def getAllowCredentials: Boolean
+  def getAllowedOrigins: HttpOriginMatcher
+  def getAllowedHeaders: HttpHeaderRange
+  def getAllowedMethods: java.lang.Iterable[HttpMethod]
+  def getExposedHeaders: java.lang.Iterable[String]
+  def getMaxAge: OptionalLong
+
+  def withAllowGenericHttpRequests(newValue: Boolean): CorsSettings
+  def withAllowCredentials(newValue: Boolean): CorsSettings
+  def withAllowedOrigins(newValue: HttpOriginMatcher): CorsSettings
+  def withAllowedHeaders(newValue: HttpHeaderRange): CorsSettings
+  def withAllowedMethods(newValue: java.lang.Iterable[HttpMethod]): CorsSettings
+  def withExposedHeaders(newValue: java.lang.Iterable[String]): CorsSettings
+  def withMaxAge(newValue: OptionalLong): CorsSettings
+}
+
+object CorsSettings {
+
+  /**
+   * Creates an instance of settings using the given Config.
+   */
+  def create(config: Config): CorsSettings = scaladsl.settings.CorsSettings(config)
+
+  /**
+   * Creates an instance of settings using the given String of config overrides to override settings set in the class
+   * loader of this class (i.e. by application.conf or reference.conf files in the class loader of this class).
+   */
+  def create(configOverrides: String): CorsSettings = scaladsl.settings.CorsSettings(configOverrides)
+
+  /**
+   * Creates an instance of CorsSettings using the configuration provided by the given ActorSystem.
+   */
+  def create(system: ActorSystem): CorsSettings = scaladsl.settings.CorsSettings(system)
+
+  /**
+   * Settings from the default loaded configuration. Note that application code may want to use the `apply()` methods
+   * instead to have more control over the source of the configuration.
+   */
+  def defaultSettings: CorsSettings = scaladsl.settings.CorsSettings.defaultSettings
```

Comment:
This `defaultSettings` was kept for backward compatibility when I worked on the [version 1.0.0](https://github.com/lomigmegard/akka-http-cors/releases/tag/v1.0.0). It's indeed not idiomatic in akka/pekko-http. 

I think it would be fine removing it (but that require more migration notes), or simply deprecating it. Looking at the downloads from `akka-http-cors_2.13`, around 1% are coming from versions 0.x. 
<img width="470" alt="Screenshot 2023-06-29 at 18 02 24" src="https://github.com/apache/incubator-pekko-http/assets/434236/c6e30099-e7ea-499b-9f79-f710c3e126ce">

Reply:
Thanks I will deprecate it then, will wait for the CI to pass

Reply:
I have just pushed the deprecations, i.e.

```scala
@deprecated("Use other CorsSettings constructors", "Akka Http Cors 1.2.0")
```

Let me know if this message makes sense, if so I will resolve the thread.

Reply:
Looks good for the message. For the "since", would it make more sense to reference the future first release of pekko? `Pekko Http Cors 1.0.0`

Reply:
Ah right, nice pickup will fix this now.